pkg/apis/machine/v1beta1/machine_webhook.go

```diff
@@ -0,0 +1,338 @@
+	clusterID, err := getClusterID()
+	if err != nil {
+		return nil, err
+	}
+
+	platform, err := getPlatform()
+	if err != nil {
+		return nil, err
+	}
+
+	return createMachineValidator(platform, clusterID), nil
+}
+
+func createMachineValidator(platform osconfigv1.PlatformType, clusterID string) *validatorHandler {
+	h := &validatorHandler{
+		clusterID: clusterID,
+	}
+
+	switch platform {
+	case osconfigv1.AWSPlatformType:
+		h.webhookOperations = validateAWS
+	default:
+		// just no-op
+		h.webhookOperations = func(h *validatorHandler, m *Machine) (bool, utilerrors.Aggregate) {
+			return true, nil
+		}
+	}
+	return h
+}
+
+// NewDefaulter returns a new defaulterHandler.
+func NewMachineDefaulter() (*defaulterHandler, error) {
+	clusterID, err := getClusterID()
+	if err != nil {
+		return nil, err
+	}
+
+	platform, err := getPlatform()
+	if err != nil {
+		return nil, err
+	}
+
+	return createMachineDefaulter(platform, clusterID), nil
+}
+
+func createMachineDefaulter(platform osconfigv1.PlatformType, clusterID string) *defaulterHandler {
+	h := &defaulterHandler{
+		clusterID: clusterID,
+	}
+
+	switch platform {
+	case osconfigv1.AWSPlatformType:
+		h.webhookOperations = defaultAWS
+	default:
+		// just no-op
+		h.webhookOperations = func(h *defaulterHandler, m *Machine) (bool, utilerrors.Aggregate) {
+			return true, nil
+		}
+	}
+	return h
+}
+
+// InjectDecoder injects the decoder.
+func (v *validatorHandler) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// InjectDecoder injects the decoder.
+func (v *defaulterHandler) InjectDecoder(d *admission.Decoder) error {
+	v.decoder = d
+	return nil
+}
+
+// Handle handles HTTP requests for admission webhook servers.
+func (h *validatorHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	m := &Machine{}
+
+	if err := h.decoder.Decode(req, m); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	klog.Infof("Validate webhook called for Machine: %s", m.GetName())
+
+	if ok, err := h.webhookOperations(h, m); !ok {
+		return admission.Denied(err.Error())
+	}
+
+	return admission.Allowed("Machine valid")
+}
+
+// Handle handles HTTP requests for admission webhook servers.
+func (h *defaulterHandler) Handle(ctx context.Context, req admission.Request) admission.Response {
+	m := &Machine{}
+
+	if err := h.decoder.Decode(req, m); err != nil {
+		return admission.Errored(http.StatusBadRequest, err)
+	}
+
+	klog.Infof("Mutate webhook called for Machine: %s", m.GetName())
+
+	if ok, err := h.webhookOperations(h, m); !ok {
+		return admission.Denied(err.Error())
+	}
+
+	marshaledMachine, err := json.Marshal(m)
+	if err != nil {
+		return admission.Errored(http.StatusInternalServerError, err)
+	}
+	return admission.PatchResponseFromRaw(req.Object.Raw, marshaledMachine)
+}
+
+func defaultAWS(h *defaulterHandler, m *Machine) (bool, utilerrors.Aggregate) {
+	klog.V(3).Infof("Defaulting aws providerSpec")
```

Comment:
I would have expected this to be in the normal capitalisation, not a strong preference though
```suggestion
	klog.V(3).Infof("Defaulting AWS providerSpec")
```